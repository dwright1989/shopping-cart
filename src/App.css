@import url('https://fonts.googleapis.com/css2?family=Vibes&display=swap');
@import url('https://fonts.googleapis.com/css2?family=Hahmlet:wght@500&display=swap');
:root{
  --main-color: white;
  --secondary-color: black;
  --tertiary-color: #c1e1ec;
}

body{
  font-family: 'Hahmlet', serif;
}

.material-symbols-outlined {
  font-variation-settings:
  'FILL' 0,
  'wght' 400,
  'GRAD' 0,
  'opsz' 48
}

.tertiary{
  color: var(--tertiary-color);
}

.page{
  color: var(--main-color);
  text-align: center;
  background-image: url("./Images/background.jpg");
}


/*NAVBAR*/
#navBar{
  display: flex;
  padding: 0vw 3rem;
  justify-content: space-between;
}

.brand{
  font-family: 'Vibes', cursive;
}

#links{
  display: flex;
  gap: 2rem;
  align-items: center;
}

#links>a{
  text-decoration: none;
  color: var(--secondary-color);
}

#links>a:hover{
  color: var(--tertiary-color);
  text-decoration: underline;
}

#basketQuantity{
  background-color: var(--tertiary-color);
  border-radius: 50px;
  width: 1.4rem;
  height: 1.4rem;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 0.8rem;
  position: absolute;
  top: 0.5rem;
  right: 2.2rem;
}

/*HOME*/
#home{
  padding: 5rem;
}
#home>h1{
  font-size: 10vw;
}

#shopButton{
  background-color: var(--tertiary-color);
  border: none;
  border-radius: 20px;
  font-size: 2vw;
  padding: 0.5vw 1vw;
}

#shopButton:hover,
#shopButton:focus{
  cursor:pointer;
  background-color: var(--main-color);
  transform: translateY(-0.25em);
}

/*ABOUT*/
#about{
  padding: 2vw;
}

#about>h1{
  font-size: 5vw;
}

#about>p{
  padding: 0 10vw;
}

/*SHOP*/
#products{
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  gap: 2rem;
  padding: 3vw 8vw;
}
#card{
  border: 1px solid var(--tertiary-color);
  background-color: var(--main-color);
  color: var(--secondary-color);
  width: 90%;
  padding: 1vw;
  display: grid;
  justify-content: space-evenly;
  align-content: space-evenly;
  border-radius: 5px;
}

#card>img{
  width: 90%;
  margin: auto;
}

#card>button{
  background-color: var(--tertiary-color);
  border: none;
  color: var(--secondary-color);
  font-family: 'Vibes', cursive;
  font-weight:bold;
  font-size: 1.5vw;
  cursor: pointer;
}

/*CART*/
#cart{
  padding: 2rem;
}
#cartContainer{
  width: 50%;
  margin: auto;
  background-color: var(--main-color);
  color: var(--secondary-color);
}

#cartItems{
  display: grid;
  gap: 1rem;
  padding: 0 2rem 2rem 2rem;
}

.cart-item{
  border: 1px solid var(--secondary-color);
}

/*FOOTER*/
#footer{
  text-align: center;
  padding-top: 1vw;
  font-size: 1vw;
}