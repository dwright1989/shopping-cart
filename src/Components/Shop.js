import Products from "../Components/Products/ProductItems";
import Card from "../Components/Card";

export default function Shop(){
    return (
        <div id="shop" className="page">
            <div id="products">
                {Products.map( (product) =>{
                    return <Card product={product}/>
                })}
            </div>
        </div>
    );
}