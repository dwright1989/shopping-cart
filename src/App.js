import {BrowserRouter, Routes, Route} from "react-router-dom";
import React from "react";
import NavBar from "./Components/NavBar.js";
import Footer from "./Components/Footer.js";
import Home from "./Components/Home.js";
import Shop from "./Components/Shop.js";
import About from "./Components/About.js";
import Cart from "./Components/Cart.js";
import './App.css';

function App() {

  const [basketQuantity, setBasketQuantity] = React.useState({
      total: 0
  });

  function handleAddToBasket(id){
      setBasketQuantity
      (prevQuantity => ({
        total: prevQuantity.total+1
    }))
  }

  

  return (
    <div id="mainContent">
      <NavBar basketQuantity={basketQuantity}/>
      <div id="pageDiv">
      <BrowserRouter>
        <Routes>
          <Route path="/" element={<Home/>}/>
          <Route path="/shop" element={<Shop  handleAddToBasket={handleAddToBasket}/>}/>
          <Route path="/about" element={<About/>}/>
          <Route path="/cart" element={<Cart/>}/>
        </Routes>
      </BrowserRouter>
      </div>    
      <Footer/>
    </div>
  );
}

export default App;
