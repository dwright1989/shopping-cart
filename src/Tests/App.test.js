import { render, renderWithContext, screen} from '@testing-library/react';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';
import App from '../App';
import Shop from '../Components/Shop';

describe('app testing', ()=>
{

    test('check renders', ()=>{
        const appComponent = renderer.create(<App/>);
        const tree = appComponent.toJSON();
        expect(tree).toMatchSnapshot();
    });
    test('handle add to basket function', async()=>{
        render(<App />);
        const shopLink = screen.getByRole('link', {name: /Products/i});
        await userEvent.click(shopLink);
        const addToBasketButton = screen.getByRole('button', {name: /Add To Basket/i});
        //userEvent.click(addToBasketButton);
        //const basketTotalP = screen.getByTestId("basket-quantity");
        //expect(basketTotalP.textContent).toBe("0");*/
      });

}
)

