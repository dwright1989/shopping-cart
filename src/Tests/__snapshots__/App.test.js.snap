// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app testing check renders 1`] = `
<div
  id="mainContent"
>
  <div
    id="navBar"
  >
    <h1
      className="brand"
    >
      Clothes
      <span
        className="tertiary"
      >
        4
      </span>
      You
    </h1>
    <div
      id="links"
    >
      <a
        href="/"
      >
        Home
      </a>
      <a
        href="shop"
      >
        Products
      </a>
      <a
        href="about"
      >
        About
      </a>
      <a
        href="cart"
      >
        <span
          className="material-symbols-outlined"
        >
          shopping_cart
        </span>
      </a>
      <p
        data-testid="basket-quantity"
        id="basketQuantity"
      >
        0
      </p>
    </div>
  </div>
  <div
    id="pageDiv"
  >
    <div
      className="page"
      id="home"
    >
      <h1
        className="brand"
      >
        Clothes
        <span
          className="tertiary"
        >
          4
        </span>
        You
      </h1>
      <p>
        A fake portfolio store
      </p>
      <button
        id="shopButton"
      >
        Shop Now
      </button>
    </div>
  </div>
  <div
    id="footer"
  >
    Created by Debbie Wright 2022
  </div>
</div>
`;
